references/cicd-pipeline/ci-config/jenkins/Jenkinsfile

```diff
@@ -124,42 +170,55 @@ pipeline {
                 -Ddeployment.description="Jenkins Build: ${env.BUILD_TAG} Author: ${env.AUTHOR_EMAIL}"
                 """
 
-          }
+          } }
         }
 
         stage('Deploy X/hybrid') {
           when {
             expression { env.API_VERSION ==  'google'}
           }
-          steps {
-            sh """
-              APIGEE_SA_TOKEN="\${APIGEE_TOKEN:-\$(gcloud auth application-default print-access-token)}"
+          steps { dir( "${env.WORK_DIR}" ) {
 
-              mvn clean install \
-                -Pgoogleapi \
-                -Denv="${env.APIGEE_ENV}" \
-                -Dtoken="\$APIGEE_SA_TOKEN" \
-                -Dorg="${env.APIGEE_ORG}" \
-                -Ddeployment.suffix="${env.APIGEE_DEPLOYMENT_SUFFIX}" \
-                -Ddeployment.description="Jenkins Build: ${env.BUILD_TAG} Author: ${env.AUTHOR_EMAIL}"
-            """
-          }
+           // Token precedence: env var; jenkins-scope sa; vm-scope sa; 
+
+            script {
+              if (env.GCP_SA_AUTH == "jenkins-scope") {
+                 withCredentials([file(credentialsId: 'apigee', variable: 'GOOGLE_APPLICATION_CREDENTIALS')]) {
+                   env.SA_TOKEN=sh(script:'gcloud auth print-access-token', returnStdout: true).trim()
+                 }
+              } else {
```

Comment:
The build failed because it tries to run `gcloud auth ..` in the pipeline runner that doesn't have a gcloud environment configured.

Proposal: can we make this an elseif that is skipped if $APIGEE_SA_TOKEN is defined?